src/Components/Aspire.Azure.AI.Projects/AzureAIProjectExtensions.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Azure.AI.Projects;
+using Aspire.Azure.Common;
+using Azure.AI.Projects;
+using Azure.Core;
+using Azure.Core.Extensions;
+using Azure.Identity;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// /// Provides extension methods for registering <see cref="AIProjectClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AzureAIProjectExtensions
+{
+    internal const string DefaultConfigSectionName = "Aspire:Azure:AI:Project";
+
+    /// <summary>
+    /// Registers <see cref="AIProjectClient"/> as a singleton in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureAIProjectSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{AIProjectClient, AIProjectClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Azure:AI:Project" section.</remarks>
```

Comment:
These references and other docs will need to be updated to, but won't repeat it everywhere.